lib/settings.h

```diff
@@ -293,6 +299,7 @@ class CPPCHECKLIB WARN_UNUSED Settings {
         std::string id = "rule"; // default id
         std::string summary;
         Severity severity = Severity::style; // default severity
+        std::shared_ptr<Regex> regex;
```

Comment:
This needs to be a shared pointer because the Settings are being copied (the copies in production code will hopefully go away soon) and we have dynamic memory.